ONSdigital/eq-questionnaire-runner
# Add support in repeating GCS for static CS with repeating answers

### What is the context of this PR?
This PR adds support for a repeating Grand Calculated Summary to include a static Calculated Summary that has repeating answers. These can be repeating block answers or dynamic answers.

This is paired with the [corresponding validator PR](https://github.com/ONSdigital/eq-questionnaire-validator/pull/227) which disallows the use of a calculated summary which has repeating answers for the same lists. So if for example, a calculated summary included a sum of dynamic answers for vehicles. In a repeating section for vehicles that calculated summary could not then be included in a repeating grand calculated summary, as it would be unclear whether to resolve the repeating answer id to the specific vehicle instance, or the collection of answers, given it was from another section. 

I had to update `test_repeating_sections_with_hub_and_spoke.json` to stop validator from breaking. It has the same answer id in the add block for both list collectors but this breaks under the new validation (and is a bug anyway) raised another card for a fix to that properly in validator

### How to review
Look over the card to check all parts have been covered. Functionally the changes are very small, its just a change to value source resolver to support returning all the items unless in a repeat for the same list. Primarily the changes are test coverage.

### How to test functionally
Use the schema `test_grand_calculated_summary_inside_repeating_section`
- Ensure that the change links in the grand calculated summaries work both for repeating answers and dynamic answers
- Make sure that changing an answer which affects the other repeating section correctly updates progress
- Note the guidance about the validation block, there is a separate card for implementing grand calculated summary dependencies. But make sure the calculated summary dependencies are working as expected.

### Checklist
